### PR TITLE
[Windows] Update the scripts to fetch latest git version

### PR DESCRIPTION
**Pull request by kishorekumar-anchala**
# Description
This PR will 
- Modify the Install-Git script to fetch latest version
#### Related issue:

- [10843](https://github.com/actions/runner-images/issues/10843)

## Check list
- [x] Related issue / work item is attached
- [ ] Tests are written (if applicable)
- [ ] Documentation is updated (if applicable)
- [ ] Changes are tested and related VM images are successfully generated

### CODE REVIEW

File: images/windows/scripts/build/Install-DACFx.ps1 [hunk 1]
```diff
@@ -5,6 +5,6 @@
 
 Install-Binary -Type MSI `
     -Url 'https://aka.ms/dacfx-msi' `
-    -ExpectedSignature 'C2048FB509F1C37A8C3E9EC6648118458AA01780'
+    -ExpectedSignature '8F985BE8FD256085C90A95D3C74580511A1DB975'
```
Comment (Alexey-Ayupov) [inline images/windows/scripts/build/Install-DACFx.ps1:8]: Should be separate PR

File: images/windows/scripts/build/Install-Git.ps1 [hunk 2]
```diff
@@ -4,18 +4,13 @@
 ##  Supply chain security: Git - checksum validation, Hub CLI - managed by package manager
 ################################################################################
 
-# Install the latest version of Git for Windows
-
-$downloadUrl = Resolve-GithubReleaseAssetUrl `
```
Comment (Alexey-Ayupov) [inline images/windows/scripts/build/Install-Git.ps1:9]: By refusing to use a function, you lose some of the functionality, e.g. download retries. Instead of rewriting the script itself, it is better to fix the `Resolve-GithubReleaseAssetUrl` function

File: images/windows/scripts/build/Install-Git.ps1 [hunk 3]
```diff
@@ -4,18 +4,13 @@
...
-    -Version "latest" `
-    -FileName (Split-Path $downloadUrl -Leaf) `
-    -HashType "SHA256"
+# Fetch the latest release info from GitHub API
+$response = Invoke-RestMethod -Uri "https://api.github.com/repos/git-for-windows/git/releases/latest"
+$asset = $response.assets | Where-Object { $_.name -match "Git-.*-64-bit.exe" }
+
+# Download URL and SHA256 checksum
+$downloadUrl = $asset.browser_download_url
+$externalHash = "83c618dcc50b63f0f7afc86a7a125169bd59559ff680683f1d915c45e05ff4cc"
```
Comment (Alexey-Ayupov) [inline images/windows/scripts/build/Install-Git.ps1:13]: There is no need to hardcode the hash if it can be obtained automatically.

**Comment by kishorekumar-anchala** [comment]
No longer needed this PR .
